services/outputhost/ackmanager.go

```diff
@@ -124,19 +124,15 @@ func (ackMgr *ackManager) getNextAckID(address int64, sequence common.SequenceNu
 	ackMgr.readLevel++ // This means that the first ID is '1'
 	ackMgr.readLevelAddr = storeHostAddress(address)
 
-	expectedReadLevel := ackMgr.levelOffset + ackMgr.readLevel
-
 	// NOTE: sequence should be zero for timer destinations, since they usually have discontinuous sequence numbers
-	if sequence != 0 && expectedReadLevel != sequence {
-		skippedMessages := sequence - expectedReadLevel
+	if sequence != 0 && sequence != ackMgr.readLevel {
```

Comment:
can't compare readLevel with sequence here. I think you need to keep track of a new member in the struct levels that tracks the current storeSeq corresponding to the ackLevelAddr. Initially, when we restore from an earlier checkpoint, ackLevelAddr will be initialized to cge.GetAckLevelOffset() and you need to initialze storeSeq to cge.GetAckLevelSeqNo(). Then, during every getNextAckID() call, compare the received storeSeq to the current storeSeq and log if they mismatch. If there is a mismatch, also reset your in-memory storeSeq to be in sync with the value received from the store.

i.e.

```

levels struct {
   ...
   readLevelAddr storeHostAddress
   readLevelStoreSeq SequenceNumber
} 

func newAckManager() {
 ....
 ackMgr.levels = &levels{
      readLevelAddr: storeHostAddress(cge.GetReadLevelOffset()),
      readLevelStoreSeq: cge.GetAckLevelSeqNo(),
 }
}

func getNextACKID(address, sequence) {
    .... 
   ackMgr.readLevel++ // this is the output seq number, always increasing by 1
    ackMgr.readLevelStoreSeq++ // this is current store seq no, can jump due to retention
    if sequence != ackMgr.readLevelStoreSeq {
             // log mismatch
             ackMgr.readLevelStoreSeq = sequence // reset here
     }
     ackMgr.readLevelAddr = address // always update the readLevelAddr

     // now store the mapping from outputSeq -> storeSeq
     ackMgr.addrs[ackMgr.readLevel] = &internalMsg{
              addr: storeHostAddress(address),
              seq:  sequence,
     }
}

```